time.go

```diff
@@ -0,0 +1,96 @@
+package null
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+)
+
+// Time is a nullable time.Time. It supports SQL and JSON serialization.
+// It will marshal to null if null.
+type Time struct {
+	Time  time.Time
+	Valid bool
+}
+
+// Scan implements the Scanner interface.
+func (t *Time) Scan(value interface{}) error {
+	t.Time, t.Valid = value.(time.Time)
+	return nil
+}
+
+// Value implements the driver Valuer interface.
+func (t Time) Value() (driver.Value, error) {
+	if !t.Valid {
+		return nil, nil
+	}
+	return t.Time, nil
+}
+
+// NewTime creates a new Time.
+func NewTime(t time.Time, valid bool) Time {
+	return Time{
+		Time:  t,
+		Valid: valid,
+	}
+}
+
+// TimeFrom creates a new Time that will always be valid.
+func TimeFrom(t time.Time) Time {
+	return NewTime(t, true)
+}
+
+// TimeFromPtr creates a new Time that will be null if t is nil.
+func TimeFromPtr(t *time.Time) Time {
+	if t == nil {
+		var ti time.Time
+		return NewTime(ti, false)
+	}
+	return NewTime(*t, true)
+}
+
+// MarshalJSON implements json.Marshaler.
+// It will encode null if this time is null.
+func (t Time) MarshalJSON() ([]byte, error) {
+	if !t.Valid {
+		return []byte("null"), nil
+	}
+	return json.Marshal(t.Time)
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+// It supports time, string and null input.
+func (t *Time) UnmarshalJSON(data []byte) error {
+	var err error
+	var v interface{}
+	json.Unmarshal(data, &v)
+	switch x := v.(type) {
+	case time.Time:
+		t.Time = x
+	case string:
+		t.Time, err = time.Parse(time.RFC3339, x)
+	case nil:
+		t.Valid = false
+		return nil
+	default:
```

Comment:
Please disregard my deleted comment.
First, it might be better to call `t.Time.UnmarshalJSON()` instead of `time.Parse()`, in case the standard library starts doing something fancy in the future. 
Second, we should check for `map[string]interface{}`. This is for backwards compatibility with storing `pq.NullTime` as its default JSON and "upgrading" to this library. I don't know if anyone uses this feature, but the other types have it. The problem is that we can't just unmarshal into t because it will cause a loop. 
Maybe something like:

```go
	case map[string]interface{}:
		t.Time.UnmarshalJSON(x["Time"].(string))
		t.Valid = x["Valid"].(bool)
```

and just let it panic on malformed data.